core/commands/shutdown.go

```diff
@@ -0,0 +1,29 @@
+package commands
+
+import (
+	"fmt"
+
+	cmds "github.com/ipfs/go-ipfs/commands"
+)
+
+var DaemonShutdownCmd = &cmds.Command{
+	Helptext: cmds.HelpText{
+		Tagline: "Shut down the ipfs daemon",
+	},
+	Run: func(req cmds.Request, res cmds.Response) {
+		nd, err := req.InvocContext().GetNode()
+		if err != nil {
+			res.SetError(err, cmds.ErrNormal)
+			return
+		}
+
+		if !nd.OnlineMode() {
```

Comment:
I am not 100% sure but I don't think this works when the daemon is started with the `--offline` flag.